res/controllers/Vestax-VCI-100MKII-scripts.js

```diff
@@ -257,32 +257,53 @@ VCI102.next_effect = function(ch, midino, value, status, group) {
     }
 };
 
-VCI102.loopSize = [4, 4, 4, 4];  // saved beatloop_size
-
 VCI102.loop = function(ch, midino, value, status, group) {
     if (value) {
         if (engine.getValue(group, "loop_enabled")) {
             engine.setValue(group, "reloop_toggle", 1);
-            // restore beatloop_size
-            engine.setValue(group, "beatloop_size", VCI102.loopSize[ch]);
         } else {
-            VCI102.loopSize[ch] = engine.getValue(group, "beatloop_size");
             engine.setValue(group, "beatloop_activate", 1);
+            engine.setValue(group, "beatloop_activate", 0);
         }
     }
 };
 
 VCI102.reloop = function(ch, midino, value, status, group) {
     if (value) {
         if (engine.getValue(group, "loop_enabled")) {
+            engine.setValue(group, "loop_out", 1);
```

Comment:
If you want to use a single button for both beatloop_activate and reloop_toggle, I suggest mapping it like I have mapped pushing the loop encoder on the Hercules P32:
  * Push with no loop active: beatloop_activate
  * Push with loop active: reloop_toggle
  * Shift + push with no loop active: reloop_toggle
  * Shift + push with loop active: reloop_andstop